plugins/scaffolder-backend/src/scaffolder/actions/builtin/fetch/helpers.ts

```diff
@@ -35,43 +35,85 @@ export async function fetchContents(options: {
 }) {
   const { reader, integrations, baseUrl, fetchUrl = '.', outputPath } = options;
 
-  let fetchUrlIsAbsolute = false;
-  try {
-    // eslint-disable-next-line no-new
-    new URL(fetchUrl);
-    fetchUrlIsAbsolute = true;
-  } catch {
-    /* ignored */
-  }
+  const fetchUrlIsAbsolute = isFetchUrlAbsolute(fetchUrl);
 
   // We handle both file locations and url ones
   if (!fetchUrlIsAbsolute && baseUrl?.startsWith('file://')) {
     const basePath = baseUrl.slice('file://'.length);
     const srcDir = resolveSafeChildPath(path.dirname(basePath), fetchUrl);
     await fs.copy(srcDir, outputPath);
   } else {
-    let readUrl;
-
-    if (fetchUrlIsAbsolute) {
-      readUrl = fetchUrl;
-    } else if (baseUrl) {
-      const integration = integrations.byUrl(baseUrl);
-      if (!integration) {
-        throw new InputError(`No integration found for location ${baseUrl}`);
-      }
-
-      readUrl = integration.resolveUrl({
-        url: fetchUrl,
-        base: baseUrl,
-      });
-    } else {
-      throw new InputError(
-        `Failed to fetch, template location could not be determined and the fetch URL is relative, ${fetchUrl}`,
-      );
-    }
+    const readUrl = getReadUrl(fetchUrl, baseUrl, integrations);
 
     const res = await reader.readTree(readUrl);
     await fs.ensureDir(outputPath);
     await res.dir({ targetDir: outputPath });
   }
 }
+
+/**
+ * A helper function that reads the content of a single file from the given URL.
+ * Can be used in your own actions, and also used behind fetch:plain:file
+ *
+ * @public
```

Comment:
```suggestion
 * Can be used in your own actions, and also used behind `fetch:plain:file`
```
We are not exporting this function so no need to mark it public :)